packages/codec/lib/basic/decode/index.ts

```diff
@@ -306,22 +310,23 @@ export function* decodeBasic(
             value: yield* decodeExternalFunction(address, selector, info)
           };
         case "internal":
-          if (strict) {
-            //internal functions don't go in the ABI!
-            //this should never happen, but just to be sure...
-            throw new StopDecodingError({
-              kind: "InternalFunctionInABIError" as const
-            });
-          }
+          //note: we no longer error here if strict is true, because we
+          //want to be able to use strict mode to decode immutables, and
+          //these can be immutables.  yes, this is a bit of an abuse of
+          //strict mode, which was meant for ABI decoding, but oh well.
```

Comment:
Oh, I see! No long *immediately* error here! It makes sense now.

Reply:
Ah, yes, sorry, I meant we no longer error at that particular point!  Do you think that's worth clarifying?

Reply:
Yes, please.

Reply:
Cool, I'll go rewrite this comment a little and then merge.